mobile/src/lib/background.js

```diff
@@ -0,0 +1,59 @@
+import { getStore } from './store'
+import { initService } from './init'
+import { logException } from './crash-reporter'
+import { loadState } from './localStorage'
+import { startMediaUpload, mediaBackup, endMediaUpload } from '../actions/mediaBackup'
+import { backupAllowed } from './network'
+
+const hasIosCordovaPlugin = () => {
+  return window.cordova.platformId === 'iOS' && window.BackgroundFetch !== undefined
+}
+
+export const launchBackground = () => {
+  if (hasIosCordovaPlugin()) {
+    launchIosBackground()
+  }
+}
+
+const launchIosBackground = () => {
+  var Fetcher = window.BackgroundFetch
+
+  const fetchCallback = () => {
+    console.log('[js] BackgroundFetch initiated')
+
+    loadState().then(persistedState => {
+      let store = getStore(persistedState)
+      initService(store)
+
+      logException('C\'est moi le Service Background!!!')
+
+      if (store.getState().mobile.settings.backupImages && backupAllowed(getState().mobile.settings.wifiOnly)) {
```

Comment:
There is something weird here as you call `store.getState()` in one hand and keep `getState()` on the other hand.